xbmc/cores/RetroPlayer/guibridge/GUIGameMessenger.h

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include <string>
+
+class CGUIComponent;
+
+namespace KODI
+{
+namespace RETRO
+{
+class CRPProcessInfo;
+class ISavestate;
+
+/*!
+ * \brief Class to send messages to the GUI, if a GUI is present
+ */
+class CGUIGameMessenger
+{
+public:
+  CGUIGameMessenger(CRPProcessInfo& processInfo);
```

Comment:
Any reason you don't pass in CGUIComponent directly?

Reply:
Encapsulation. Caller doesn't need to know about the GUI.

Reply:
That's what CRenderContext is for, hiding globals like CGUIComponent.

I wanted to pass a CRenderContext here, but CProcessInfo owns the CRenderContext, so we might as well pass CProcessInfo in case the messenger needs access to a sister member of CRenderContext in the future.

You can also see when constructed in CRetroPlayer, the other subsystems nearby are given CProcessInfo.